homeassistant/components/voice_assistant/pipeline.py

```diff
@@ -92,6 +98,37 @@ def finish(self):
             )
         )
 
+    async def speech_to_text(
+        self, metadata: stt.SpeechMetadata, stream: AsyncIterable[bytes]
+    ) -> stt.SpeechResult:
+        """Run text to speech portion of pipeline. Returns URL of TTS audio."""
+        self.event_callback(
+            PipelineEvent(
+                PipelineEventType.STT_START,
+                {
+                    "engine": self.pipeline.stt_engine or "default",
+                },
+            )
+        )
+
+        stt_provider = stt.async_get_provider(self.hass, self.pipeline.stt_engine)
+        result = await stt_provider.async_process_audio_stream(metadata, stream)
```

Comment:
Should we catch exceptions here.

Reply:
Catching them now in each part of the pipeline and returning a relevant error.